lib/formats/bs_format_ripejson.c

```diff
@@ -220,6 +226,49 @@ static int process_common_fields(bgpstream_format_t *format,
   return 0;
 }
 
+// readline
+static int readline(bgpstream_format_t *format)
+{
+  int total_read = 0;
+  int current_read = 0;
+  char *buffer_ptr = STATE->json_string_buffer;
+
+  while(1){
+    // read line, including linebreak
+    current_read = bgpstream_transport_readline(
+      format->transport, buffer_ptr, BUFSIZE - total_read, 0);
+    total_read+=current_read;
+
+    if(current_read == 0){
+      // reach EOF, no bytes read, return 0
+      return total_read;
+    }
+
+    if(current_read < BUFSIZE - total_read -1 ||
+      buffer_ptr[current_read-1] == '\n'){
+      // if read less than buffer, or last byte is an newline, readline is finished
+
+      // replace linebreak with null
+      if(buffer_ptr[current_read-1] == '\n'){
```

Comment:
- [ ] Change `if` to `assert`
- [ ] add comments to document the assumption that input should be always newline terminated